File: Telegram/NotificationService/NotificationServiceObjC/Sources/Serialization.m
#import "Serialization.h"

@implementation Serialization

- (NSUInteger)currentLayer {
    return 133;
}

- (id _Nullable)parseMessage:(NSData * _Nullable)data {
    return nil;
}

- (MTExportAuthorizationResponseParser _Nonnull)exportAuthorization:(int32_t)datacenterId data:(__autoreleasing NSData **)data {
    return ^MTExportedAuthorizationData *(NSData *resultData) {
        return nil;
    };
}

- (NSData * _Nonnull)importAuthorization:(int64_t)authId bytes:(NSData * _Nonnull)bytes {
    return [NSData data];
}

- (MTRequestDatacenterAddressListParser)requestDatacenterAddressWithData:(__autoreleasing NSData **)data {
    return ^MTDatacenterAddressListData *(NSData *resultData) {
        return nil;
    };
}

- (MTRequestNoopParser)requestNoop:(__autoreleasing NSData **)data {
    return ^id(NSData *resultData) {
        return nil;
    };
}

@end

File: NotificationService/Serialization.swift
import Foundation
import MtProtoKitDynamic

public class BoxedMessage: NSObject {
    public let body: Any
    public init(_ body: Any) {
        self.body = body
    }
}

public class Serialization: NSObject, MTSerialization {
    public func currentLayer() -> UInt {
        return 97
    }
    
    public func parseMessage(_ data: Data!) -> Any! {
        return nil
    }
    
    public func exportAuthorization(_ datacenterId: Int32, data: AutoreleasingUnsafeMutablePointer<NSData?>) -> MTExportAuthorizationResponseParser!
    {
        return { data -> MTExportedAuthorizationData? in
            return nil
        }
    }
    
    public func importAuthorization(_ authId: Int32, bytes: Data!) -> Data! {
        return Data()
    }
    
    public func requestDatacenterAddress(with data: AutoreleasingUnsafeMutablePointer<NSData?>) -> MTRequestDatacenterAddressListParser! {
        return { response -> MTDatacenterAddressListData? in
            return nil
        }
    }
    
    public func requestNoop(_ data: AutoreleasingUnsafeMutablePointer<NSData?>!) -> MTRequestNoopParser! {
        return { response -> AnyObject? in
            return nil
        }
    }
}
